### PR TITLE
ci(changelog): squashed changelog directory

### PR DESCRIPTION
**Pull request by gunnar-solo**
There is an issue with verifying this, since changelog bot cannot handle more than 500 file changes at once

To verify this, I first attempted to create a branch off of this one: `squash-changelogs-poc`, then open a [pull request](https://github.com/solo-io/gloo/pull/7136) against this branch.  UNFORTUNATELY, this did not proc changelog bot--presumably because it is not a main branch.

as a secondary, less-optimal way of verifying, I have created a [simulated repo](https://github.com/solo-io/actions-integration-testing/pull/6) which does the same operation that this PR is performing

----------

It is _quite hard_ to use the `Files changed` tab of this PR, due to the excessive amount of changes.  So, here are the highlights:
* edits to `hack/squash-changelogs.sh` to change top-level folders to `v1.13.x` format, as opposed to `v1.13`
* added `activeSubdirectory: "v1.13.x"` field to `changelog/validation.yaml` to opt in to new feature
* added `activeVersionSubdirectory: "1.12.x"` as part of a repair to the `hack/find_latest_enterprise_version.go` script
* ran `hack/squash-changelogs.sh` (moved a bunch of changelogs to nested folders)

----------
lastly, for visual reference, the final result:
Screen Shot 2022-09-12 at 3 44 36 PM

### CODE REVIEW

**Comment by gunnar-solo** [comment]
/skip-changelog <-- unfortunately, this is necessary, because we have too many changed files for changelog bot to handle/parse as it is currently written

**Comment by gunnar-solo** [comment]
/skip-changelog <-- unfortunately, this is necessary, because we have too many changed files for changelog bot to handle/parse as it is currently written

**Comment by gunnar-solo** [comment]
/skip-changelog <-- unfortunately, this is necessary, because we have too many changed files for changelog bot to handle/parse as it is currently written

File: changelog/1.13.x/v1.13.0-beta13/squash-changelogs.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+changelog:
```
Comment (jenshu) [inline changelog/1.13.x/v1.13.0-beta13/squash-changelogs.yaml:1]: should be beta15 now

File: changelog/validation.yaml [hunk 2]
```diff
@@ -1,3 +1,9 @@
 allowedLabels:
   - rc
-  - beta
\ No newline at end of file
+  - beta
+activeSubdirectory: "v1.13.x"
```
Comment (jenshu) [inline changelog/validation.yaml:4]: why does this have a `v` if the folders don't?
also what was the reasoning for not having the top-level folders have a `v`?

**Pull request closed without merging**